server/keyshare/keyshareserver/postgresdb.go

```diff
@@ -206,9 +211,9 @@ func (db *postgresDB) addLog(user *User, eventType eventType, param interface{})
 	return err
 }
 
-func (db *postgresDB) addEmailVerification(user *User, emailAddress, token string) error {
-	expiry := time.Now().Add(emailTokenValidity * time.Hour)
-	maxPrevExpiry := expiry.Add(-1 * emailTokenRateLimitDuration * time.Minute)
+func (db *postgresDB) addEmailVerification(user *User, emailAddress, token string, validity int) error {
+	expiry := time.Now().Add(time.Duration(validity) * time.Hour)
+	maxPrevExpiry := expiry.Add(-1 * time.Duration(emailTokenRateLimitDuration) * time.Minute)
```

Comment:
More like a fun fact if you were wondering what was happening here:
In `time.Duration(emailTokenRateLimitDuration)` it is not really required to add the `time.Duration(...)` part.
Something like `maxPrevExpiry := expiry.Add(-1 * emailTokenRateLimitDuration * time.Minute)` would work too, because `time.Duration` itself is also represented as an `int`. This is the reason why it worked in the old situation.